src/pages/ReportDetailsPage.tsx

```diff
@@ -263,6 +264,25 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
     const shouldShowWriteCapability = !isMoneyRequestReport;
     const shouldShowMenuItem = shouldShowNotificationPref || shouldShowWriteCapability || (!!report?.visibility && report.chatType !== CONST.REPORT.CHAT_TYPE.INVOICE);
 
+    const isPayer = ReportUtils.isPayer(session, moneyRequestReport);
+    const isSettled = ReportUtils.isSettled(moneyRequestReport?.reportID ?? '-1');
+
+    const shouldShowCancelPaymentButton = caseID === CASES.MONEY_REPORT && isPayer && isSettled && ReportUtils.isExpenseReport(moneyRequestReport);
```

Comment:
Should we make a utility function out of this?

Reply:
Also, why are we adding this extra check here? This did not exist in the `MoneyReportHeader` component previously.

Reply:
I don't think that is necessary, this is done in the same way as `shouldShowLeaveButton` and `canUnapproveRequest` have been handled in the same file.

Reply:
Again, the extra check is due to `ReportDetailsPage` unifying the functionality of three different components from before, `MoneyRequestHeader`, `MoneyReportHeader` and `HeaderView`. I wanted to make sure this only works for the case which it comes from, hence the `CASES.MONEY_REPORT` check.